consume-more-stuff/server/routes/categories.js

```diff
@@ -1,8 +1,129 @@
 const express = require("express");
 const router = express.Router();
+const Category = require("../db/Models/Category");
+const Item = require("../db/Models/Item");
 
 router.get("/", (req, res) => {
-  res.send(`smoke test - categories`);
+  return Category.fetchAll()
+    .then(cats => {
+      res.json(cats);
+    })
+    .catch(err => console.error(err));
+});
+
+router.get("/home", (req, res) => {
+  return Item.fetchAll({ withRelated: ["category_id", "condition_id", "itemStatus_id"] })
+    .then(items => {
+      const itemsObjs = items.toJSON();
+      let autoArr = [];
+      let clothArr = [];
+      let elecArr = [];
+      let genArr = [];
+      let servArr = [];
+
+      itemsObjs.forEach(element => {
+        const catId = element.category_id;
+
+        if (catId.id === 1 && autoArr.length <= 5) {
+          autoArr.push(element);
+        } else if (catId.id === 2 && clothArr.length <= 5) {
+          clothArr.push(element);
+        } else if (catId.id === 3 && elecArr.length <= 5) {
+          elecArr.push(element);
+        } else if (catId.id === 4 && genArr.length <= 5) {
+          genArr.push(element);
+        } else if (catId.id === 5 && servArr.length <= 5) {
+          servArr.push(element);
+        } else {
+          return "error has occurred";
+        }
+      });
+      return res.send({ autoArr, clothArr, elecArr, genArr, servArr });
+    })
+    .catch(err => {
+      console.error(err);
+      return res.status(400).send(`An error occurred`);
```

Comment:
A general server error would return a `500`. A `400` is for unable to understand a response of some paramaters.